### PR TITLE
add template and codes tables for HEALPix grid

### PR DESCRIPTION
**Pull request by amilan17**
#197

### CODE REVIEW

File: GRIB2_CodeFlag_3_1_CodeTable_en.csv [hunk 1]
```diff
@@ -42,9 +42,13 @@ Grid definition template number,,102-109,,Reserved,,,,Operational
 Grid definition template number,,110,,Equatorial azimuthal equidistant projection,,,,Operational
 Grid definition template number,,111-119,,Reserved,,,,Operational
 Grid definition template number,,120,,Azimuth-range projection,,,,Operational
-Grid definition template number,,121-139,,Reserved,,,,Operational
+Grid definition template number,,121-129,,Reserved,,,,Operational
```
Comment (amilan17) [inline GRIB2_CodeFlag_3_1_CodeTable_en.csv:45]: Need to revert. 130 is replaced by 150.

File: GRIB2_Template_3_150_GridDefinitionTemplate_en.csv [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Title_en,OctetNo,Contents_en,Note_en,noteIDs,codeTable,flagTable,Status
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",15,Shape of the Earth,(see Code table 3.2),,3.2,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",16,Scale factor of radius of spherical Earth,,,,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",17-20,Scaled value of radius of spherical Earth,,,,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",21,Scale factor of major axis of oblate spheroid Earth,,,,,Operational
```
Comment (amilan17) [inline GRIB2_Template_3_150_GridDefinitionTemplate_en.csv:5]: @sebvi octets 21-30 differ from the table in the issue. 

e.g. "Scale factor of Earth major axis" != "Scale factor of major axis of oblate spheroid Earth"

File: GRIB2_Template_3_150_GridDefinitionTemplate_en.csv [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",22-25,Scaled value of major axis of oblate spheroid Earth,,,,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",26,Scale factor of minor axis of oblate spheroid Earth,,,,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",27-30,Scaled value of minor axis of oblate spheroid Earth,,,,,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",31,Resolution and component flags,(see Flag table 3.3),,,3.3,Operational
+"Hierarchical Equal Area isoLatitude Pixelization grid (HEALPix)",32-35,nsides - number of sides within a rhomboid shape,(see note 1),197a,,,Operational
```
Comment (amilan17) [inline GRIB2_Template_3_150_GridDefinitionTemplate_en.csv:10]: add notes to notes file

File: GRIB2_CodeFlag_3_12_CodeTable_en.csv
Comment (amilan17) [inline GRIB2_CodeFlag_3_12_CodeTable_en.csv:1]: @sebvi branch adds "or points ordering" to name of this table. Do you want to keep it?

File: GRIB2_CodeFlag_3_13_FlagTable_en.csv
Comment (amilan17) [inline GRIB2_CodeFlag_3_13_FlagTable_en.csv:1]: @sebvi add notes from issue summary?

**Review by amilan17 — COMMENTED** [review]
@sebvi a couple comments for your to respond to. Thx.

File: GRIB2_Template_3_150_GridDefinitionTemplate_en.csv (hunk 2 above)
Comment (sebvi) [reply GRIB2_Template_3_150_GridDefinitionTemplate_en.csv:5]: I have use for the template the same wording than in all other section 3 template. I could update the issue description to reflect that if really necessary.

File: GRIB2_CodeFlag_3_1_CodeTable_en.csv (hunk 1 above)
Comment (sebvi) [reply GRIB2_CodeFlag_3_1_CodeTable_en.csv:45]: adding as 3.150 instead of 3.130

File: GRIB2_Template_3_150_GridDefinitionTemplate_en.csv (hunk 3 above)
Comment (sebvi) [reply GRIB2_Template_3_150_GridDefinitionTemplate_en.csv:10]: done

**Comment by sebvi** [comment]
I still need to add table wide bnotes, not to specific entry in a table. How is this done?

File: GRIB2_CodeFlag_3_13_FlagTable_en.csv
Comment (amilan17) [reply GRIB2_CodeFlag_3_13_FlagTable_en.csv:1]: I added notes. iss197a-iss197c. 

(1) see appendix XXX  
(2) In the Ring ordering scheme, the 12 rhomboids are not considered and the scanning mode applies to the grid in its entirety.  
(3) In the Nested ordering scheme, the 12 rhomboids are scanned in order defined in Gorsky et al. 2015. The flag bits then refer to how the points are scanned within a rhomboid.

**Pull request merged**